LabAnswers/src/MyLinkedList.java

```diff
@@ -0,0 +1,118 @@
+import java.util.Iterator;
+
+public class MyLinkedList<T> implements Iterable<T>{
+
+	private Node<T> head;
+	private int size = 0;
+
+	public int getSize() {
+		return size;
+	}
+
+	public boolean remove(int index) {
+		if (index > size || index < 0) {
+			return false;
+		}
+		Node<T> nodeBefore = head;
+		Node<T> nodeAfter;
+		for (int i = 0; i < index - 1; i++) {
+			nodeBefore = nodeBefore.getNext();
+		}
+		nodeAfter = nodeBefore.getNext().getNext();
+		nodeBefore.setNext(nodeAfter);
+		size--;
+		return true;
+
+	}
+
+	public T get(int index) {
+		if (index>size || index<0){return null;}
+		Node<T> node = head;
+		for (int i=0; i<index; i++){
+			node = node.getNext();
+		}
+		return node.getData();
+	}
+
+	public void add(T obj) {
+		if(size == 0){
+			head = new Node<>(obj);
+			size++;
+			return;
+		}
+		Node<T> last = head;
+		Node<T> afterLast = last.getNext();
+		while (afterLast!=null){
```

Comment:
Your condition here could be `last.getNext() != null`, then you wouldn't need `afterLast` at all